Services/Administration/OJS.Services.Administration.Business/Implementations/ProblemsBusinessService.cs

```diff
@@ -176,6 +179,32 @@ public async Task<bool> UserHasProblemPermissions(int problemId, string? userId,
             return await this.contestsBusiness.UserHasContestPermissions(problem.ContestId, userId, isUserAdmin);
         }
 
+        public async Task ReevaluateProblemsByOrderBy(int contestId, int problemId)
+        {
+            var problemsGroups = this.problemGroupData.GetAllNonDeletedByContest(contestId);
+
+            if (problemsGroups.All(pg => pg.Problems.Count == 1
+                                         && !pg.Problems.Single().IsDeleted))
```

Comment:
Why is this !pg.Problems.Single().IsDeleted?

Reply:
I am using it because there could be ProblemGroups with deleted problem in OnsitePracticalExam. In order to prevent reevaluation of all problemsGroups and Problems, I added an additional check. Please let me know, if this solution is acceptable or not.

Reply:
If the group has 2 problems - 1 deleted and one actual, we want to reevaluate, but this check will exclude such scenario, as pg.Problems.Count == 1 will return false. We want to just modify the check to
`pg.Problems.Count(p => !p.IsDeleted) == 1` and this will be enough, no need for Single()